themes/default/content/blog/reducing-our-error-rate/index.md

```diff
@@ -0,0 +1,77 @@
+
+- Checking the issue tracker for a pre-existing ticket to add additional context, a +1, etc.
+- Reaching out to a teammate who just shipped some buggy code so they can get a fix out
+- Filing a well-documented bug
+- Starting a slack thread with the team to raise a known issue that seems to be cropping up more regularly
+- Just opening a PR if it’s a simple fix
+
+Rember we only have 250 hours or 15,000 minutes per year to triage with a single on-call engineer. At five minutes a pop we can triage ~3000 errors per year until things start dropping on the floor.
+
+Now imagine you work on a new product and you just launched your MVP. It is early days, but there is some traction and you’re seeing API traffic at a rate of 1,000,000 requests per year. Your annual budget for triaging as we previously determined is 3000 errors, which yields a maximum permissible error rate of 0.3%, or a 99.7% success rate.
+
+Over the next six months, the team iterates, listens to customers, and delivers a ton of value. As a result, traffic levels grow to a rate of 10,000,000 requests per year. But still, our team hasn’t grown much and we still have just one engineer on call at any given point in time meaning our error triage capacity remains fixed at 3000 errors per year. In order to keep up with triaging the error stream at the increased levels of traffic, the team must improve the error rate from 0.3% to 0.03%. And hopefully, this team continues to be successful, increasing API traffic superlinearly in years to come.
+
+If you want to be able to read every error message, then the error rate has to come down as API traffic increases.
+
+## Why Should You Care?
+
+Pulumi aspires to be the most reliable infrastructure that our customers interact with, and the benefits towards that end are reason enough for us. But this process is by no means free, and there is always an opportunity cost.
+
+We noticed a powerful second-order effect emerge over time. The team began obsessing over the user experience. Following this process builds a visceral understanding of how your system behaves. You know when a new feature has a bug before the first support tickets get opened. You know which customer workloads will require scaling investments in the next 90 days. You know what features see heavy usage and which ones customers ignore. You’re forced to confront every wart in your application. Slowly, your team builds a better understanding of your customers and this trickles down into every aspect of product development. You begin to hold yourselves to higher standards.
+
+A friend and engineer at a large-cap software company read a draft of this post and told me:
+
+ > “I can’t imagine this process being set on any of the systems I work on at *REDACTED*. They are so prone to 500s and everybody shrugs their shoulders as if it is just commonplace.”
+
+Apathy in place of customer obsession is not an option if you are a startup that wants to disrupt anything.
```

Comment:
> if you are a startup that wants to disrupt anything

I'm sure we don't mean to say it this way, but the implication here is that _we are_ a startup that _does_ want to disrupt something (as opposed to just build a great product that people love), and that disruption in general is what everyone wants. I don't think everyone would agree, though, and I'd be somewhat concerned that this might come across as a bit arrogant and inadvertently reflect poorly on us -- although I wouldn't mind hearing what others think on this, in case I'm just being too sensitive. 😄